### PR TITLE
Fixed validation for non required fields

### PR DESCRIPTION
**Pull request by tazo90**
This unlocks validation for not required fields.

### CODE REVIEW

**Comment by tazo90** [comment]
@lsbardel thanks for suggestions.

I will create tests for this. 
Also I will make fix for [https://github.com/quantmind/lux/issues/186](https://github.com/quantmind/lux/issues/186) in this pull request.

**Comment by lsbardel** [comment]
great thanks

**Comment by tazo90** [comment]
This fixes following issues:

1. personal email validation does not display
2. [https://github.com/quantmind/lux/issues/186](https://github.com/quantmind/lux/issues/186)
     - if exist validation message from server side then display it
     - else display angular validation

**Comment by lsbardel** [comment]
can you rerun the tests?

**Comment by tazo90** [comment]
sorry, test are passed now.

File: luxjs/tests/forms/form.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            expect(tags[0].tagName).toBe('LABEL');
+            expect(tags[1].tagName).toBe('INPUT');
+
+            expect(tags[1].required).toBe(true);
+            expect(tags[1].type).toBe('email');
+            expect(tags[1].name).toBe('login');
+        }));
+
+        it("on success, email validate", inject(function($compile, $rootScope) {
+            lux.formTests.simple = testFormUtils.createForm({
```
Comment (lsbardel) [inline luxjs/tests/forms/form.js]: use a different attribute than ``simple``, don't reuse the same attribute as the previous test. ``lux`` is a global variable.

File: luxjs/tests/forms/form.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            var form = angular.element(element).find('form');
+            scope = form.scope();
+            scope.form.login.$setViewValue('user@example.com');
+
+            expect(scope.form.login.$valid).toBe(true);
+            expect(scope.form.login.$invalid).toBe(false);
+        }));
+
+        it("on failure, email validate", inject(function($compile, $rootScope) {
+            lux.formTests.simple = testFormUtils.createForm({
```
Comment (lsbardel) [inline luxjs/tests/forms/form.js]: same here, use a different name, ``lux.formTests.scscds``

File: luxjs/tests/forms/form.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            var form = angular.element(element).find('form');
+            scope = form.scope();
+            scope.form.login.$setViewValue('example.com');
+
+            expect(scope.form.login.$valid).toBe(false);
+            expect(scope.form.login.$invalid).toBe(true);
+        }));
+
+        it('submit empty email field', inject(function($compile, $rootScope) {
+            lux.formTests.simple = testFormUtils.createForm({
```
Comment (lsbardel) [inline luxjs/tests/forms/form.js]: same here

File: luxjs/tests/forms/form.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            scope.form.login.$setViewValue('');
+            scope.form.$setSubmitted(true);
+            scope.$digest();
+            expect(validators.eq(0).hasClass('ng-hide')).toBe(false);
+            expect(validators.eq(1).hasClass('ng-hide')).toBe(true);
+            expect(validators.eq(2).hasClass('ng-hide')).toBe(true);
+        }));
+
+        it('submit invalid email field', inject(function($compile, $rootScope) {
+            lux.formTests.simple = testFormUtils.createForm({
```
Comment (lsbardel) [inline luxjs/tests/forms/form.js]: and here

**Comment by lsbardel** [comment]
no they are not passing. Maybe you want to wait for the build to finish before announcing that tests pass?

**Comment by tazo90** [comment]
It was weird, on my machine test was ok but in travis not.

It's ok now.

**Comment by lsbardel** [comment]
:+1: good work @tazo90 
the reason it was failing randomly was the use of the same global variable for forms.

**Pull request merged**